### ISSUE TITLE
Allow test-kitchen 2.0.0

### ISSUE THREAD

**Issue by baurmatt** [neillturner/kitchen-puppet#205]
Since test-kitchen 2.0.0 was released last night, our CI builds fail because we depend on the current version (3.4.2) of kitchen-puppet but bundle installs 3.1.2. The reason for this is, that since 3.1.3 the test-kitchen dependency is `~> 1.4`.

Example Gemfile
```ruby
source 'https://rubygems.org'

gem 'kitchen-puppet'
gem 'test-kitchen'
```

bundle install log:
<details><summary>Collapsed log</summary>
<p>

```bash
$ bundle install --jobs $(nproc)
Fetching gem metadata from https://rubygems.org/.........
Using bundler 1.16.6
Fetching erubis 2.7.0
Fetching builder 3.2.3
Fetching ffi 1.10.0
Installing builder 3.2.3
Installing erubis 2.7.0
Fetching httpclient 2.8.3
Installing ffi 1.10.0 with native extensions
Installing httpclient 2.8.3
Fetching net-ssh 5.2.0
Installing net-ssh 5.2.0
Fetching mixlib-shellout 2.4.4
Installing mixlib-shellout 2.4.4
Fetching mixlib-versioning 1.2.7
Fetching thor 0.20.3
Installing mixlib-versioning 1.2.7
Installing thor 0.20.3
Fetching little-plugger 1.1.4
Installing little-plugger 1.1.4
Fetching multi_json 1.13.1
Fetching nori 2.6.0
Installing multi_json 1.13.1
Installing nori 2.6.0
Fetching rubyntlm 0.6.2
Fetching rubyzip 1.2.2
Installing rubyntlm 0.6.2
Installing rubyzip 1.2.2
Fetching gyoku 1.3.1
Installing gyoku 1.3.1
Fetching net-scp 1.2.1
Fetching net-ssh-gateway 2.0.0
Installing net-scp 1.2.1
Fetching mixlib-install 3.11.11
Installing net-ssh-gateway 2.0.0
Fetching logging 2.2.2
Installing mixlib-install 3.11.11
Installing logging 2.2.2
Fetching gssapi 1.2.0
Installing gssapi 1.2.0
Fetching winrm 2.3.1
Installing winrm 2.3.1
Fetching winrm-fs 1.3.2
Installing winrm-fs 1.3.2
Fetching winrm-elevated 1.1.1
Installing winrm-elevated 1.1.1
Fetching test-kitchen 2.0.0
Installing test-kitchen 2.0.0
Fetching kitchen-puppet 3.1.2
Installing kitchen-puppet 3.1.2
Bundle complete! 2 Gemfile dependencies, 25 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
```

</p>
</details>

I would suggest upgrading the test-kitchen dependency to `'>= 1.4', '< 3.0'`. I will provide a PR for this.

**Comment by baurmatt** [comment]
Also:

```
mbaur(2019-03-21 14:07:09)  ~/Sources/kitchen-puppet  (master)
$ bundle install --jobs $(nproc)
Fetching gem metadata from https://rubygems.org/........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Bundler could not find compatible versions for gem "test-kitchen":
  In snapshot (Gemfile.lock):
    test-kitchen (= 2.0.0)

  In Gemfile:
    kitchen-docker (~> 2.6.0) was resolved to 2.6.0, which depends on
      test-kitchen (>= 1.0.0)

    kitchen-puppet was resolved to 3.4.2, which depends on
      test-kitchen (~> 1.4)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

**Comment by neillturner** [comment]
resolved in latest version of kitchen-puppet 
closed
